coral-trino/src/test/java/com/linkedin/coral/trino/rel2trino/HiveToTrinoConverterTest.java

```diff
@@ -359,6 +359,29 @@ public void testIfWithNullAsThirdParameter() {
     assertEquals(expandedSql, targetSql);
   }
 
+  @Test
+  public void testSelectNullFields() {
+    RelNode relNode = hiveToRelConverter.convertSql("SELECT NULL, NULL AS TMP, CAST(NULL AS BINARY)");
+    String targetSql =
+        "SELECT NULL, NULL AS \"TMP\", CAST(NULL AS VARBINARY)\n" + "FROM (VALUES  (0)) AS \"t\" (\"ZERO\")";
+
+    RelToTrinoConverter relToTrinoConverter = new RelToTrinoConverter();
+    String expandedSql = relToTrinoConverter.convert(relNode);
+    assertEquals(expandedSql, targetSql);
+  }
+
+  @Test
+  public void testRelationalOperators() {
+    RelNode relNode = hiveToRelConverter.convertSql("SELECT 1 > 2 AS col1, 2 <= 3 AS col2 WHERE 3=3");
+    String targetSql =
+        "SELECT TRY_CAST(1 AS VARCHAR) > TRY_CAST(2 AS VARCHAR) AS \"col1\", TRY_CAST(2 AS VARCHAR) <= TRY_CAST(3 AS VARCHAR) AS \"col2\"\n"
+            + "FROM (VALUES  (0)) AS \"t\" (\"ZERO\")\n" + "WHERE TRY_CAST(3 AS VARCHAR) = TRY_CAST(3 AS VARCHAR)";
```

Comment:
I am worried about the query result regression if we add `try_cast` for some operators like `>`/`<`.
For example:
HQL:
```
SELECT * FROM t WHERE 10 > 6; // always true
```
Translated Trino SQL:
```
SELECT * FROM t WHERE try_cast(10, varchar) > try_cast(6, varchar); // always false because '10' < '6'
```
This kind of regression can't be caught by regression test, but will affect the query result.
cc: @wmoustafa

Reply:
Good catch. The agreement was to fix this behavior once the type inference work is checked in. Now that it is checked in, maybe we can fix it as part of this PR.

Reply:
@ljfgem @wmoustafa  I have updated the PR to address this concern. PTAL

Reply:
Thanks @aastha25, have you run the regression test again for this update? Let's ensure there is no translation failure.